crates/oxc_transformer/src/plugins/module_runner_transform.rs

```diff
@@ -0,0 +1,2055 @@
+    use oxc_parser::Parser;
+    use oxc_semantic::SemanticBuilder;
+    use oxc_span::SourceType;
+    use oxc_tasks_common::print_diff_in_terminal;
+    use oxc_traverse::traverse_mut;
+
+    use crate::{JsxOptions, JsxRuntime, TransformOptions, context::TransformCtx, jsx::Jsx};
+
+    use super::ModuleRunnerTransform;
+
+    fn transform(source_text: &str, is_jsx: bool) -> Result<String, Vec<OxcDiagnostic>> {
+        let source_type = SourceType::default().with_jsx(is_jsx);
+        let allocator = Allocator::default();
+        let ret = Parser::new(&allocator, source_text, source_type).parse();
+        let mut program = ret.program;
+        let (mut symbols, mut scopes) =
+            SemanticBuilder::new().build(&program).semantic.into_symbol_table_and_scope_tree();
+        if is_jsx {
+            let mut jsx_options = JsxOptions::enable();
+            jsx_options.runtime = JsxRuntime::Classic;
+            jsx_options.jsx_plugin = true;
+            let ctx = TransformCtx::new(Path::new(""), &TransformOptions::default());
+            let mut jsx = Jsx::new(jsx_options, None, AstBuilder::new(&allocator), &ctx);
+
+            (symbols, scopes) = traverse_mut(&mut jsx, &allocator, &mut program, symbols, scopes);
+        }
+        let mut module_runner_transform = ModuleRunnerTransform::default();
+        traverse_mut(&mut module_runner_transform, &allocator, &mut program, symbols, scopes);
+
+        if !ret.errors.is_empty() {
+            return Err(ret.errors);
+        }
+        let code = CodeGenerator::new()
+            .with_options(CodegenOptions { single_quote: true, ..CodegenOptions::default() })
+            .build(&program)
+            .code;
+        Ok(code)
+    }
+
+    fn format_expected_code(source_text: &str) -> String {
+        let source_type = SourceType::default();
+        let allocator = Allocator::default();
+        let ret = Parser::new(&allocator, source_text, source_type).parse();
+
+        CodeGenerator::new()
+            .with_options(CodegenOptions { single_quote: true, ..CodegenOptions::default() })
+            .build(&ret.program)
+            .code
+    }
+
+    fn test_same(source_text: &str, expected: &str) {
+        let expected = format_expected_code(expected);
+        let result = transform(source_text, false).unwrap();
+        if result != expected {
+            let diff = TextDiff::from_lines(&expected, &result);
+            print_diff_in_terminal(&diff);
+            panic!("Expected code does not match the result");
+        }
+    }
+
+    fn test_same_jsx(source_text: &str, expected: &str) {
+        let expected = format_expected_code(expected);
+        let result = transform(source_text, true).unwrap();
+        if result != expected {
+            let diff = TextDiff::from_lines(&expected, &result);
+            print_diff_in_terminal(&diff);
+            panic!("Expected code does not match the result");
+        }
+    }
+
+    #[test]
+    fn default_import() {
```

Comment:
If we are going to support `.deps` / `.dynamicDeps`, I think we need to add tests for it (`ssrTransform.spec.ts` has them).

Reply:
I am prefer not to support them. See https://github.com/oxc-project/oxc/pull/9371#discussion_r1974791415